pkg/controller/istiocsr/serviceaccounts.go

```diff
@@ -0,0 +1,60 @@
+package istiocsr
+
+import (
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/openshift/cert-manager-operator/api/operator/v1alpha1"
+	"github.com/openshift/cert-manager-operator/pkg/operator/assets"
+)
+
+func (r *Reconciler) createOrApplyServiceAccounts(istiocsr *v1alpha1.IstioCSR, resourceLabels map[string]string, istioCSRCreateRecon bool) error {
+	desired := r.getServiceAccountObject(istiocsr, resourceLabels)
+
+	serviceAccountName := fmt.Sprintf("%s/%s", desired.GetNamespace(), desired.GetName())
+	r.log.V(1).Info("reconciling serviceaccount resource", "name", serviceAccountName)
+	fetched := &corev1.ServiceAccount{}
+	key := types.NamespacedName{
+		Name:      desired.GetName(),
+		Namespace: desired.GetNamespace(),
+	}
+	exist, err := r.Exists(r.ctx, key, fetched)
+	if err != nil {
+		return FromClientError(err, "failed to check %s serviceaccount resource already exists", serviceAccountName)
+	}
+
+	if exist {
+		if istioCSRCreateRecon {
```

Comment:
any specific reason to differ from the usual `if exist && istioCSRCreateRecon {`
(which is all all the other `createOrApply*` funcs), ofc less nesting would improve readability.

Reply:
Instead of using an else for the log statement, broke this into two `if`s

Reply:
dint get why serviceaccounts be treated differently from the other resources, in this case?